api/handlers/exporter_test.go

```diff
@@ -465,3 +471,687 @@ func TestExporterDecideds_ErrorGetParticipantsInRange(t *testing.T) {
+				copy(pk[:], pkBytes)
+
+				validatorStore.ValidatorByIndexFunc = func(index phase0.ValidatorIndex) (*ssvtypes.SSVShare, bool) {
+					share := &ssvtypes.SSVShare{}
+					copy(share.ValidatorPubKey[:], pkBytes)
+					return share, true
+				}
+
+				store.GetValidatorDutyFunc = func(role spectypes.BeaconRole, slot phase0.Slot, pubkey spectypes.ValidatorPK) (*dutytracer.ValidatorDutyTrace, error) {
+					return &dutytracer.ValidatorDutyTrace{
+						ValidatorDutyTrace: exportertypes.ValidatorDutyTrace{
+							Slot:      150,
+							Role:      role,
+							Validator: 1,
+						},
+					}, nil
+				}
+			},
+			expectedStatus: http.StatusOK,
+			validateResp: func(t *testing.T, rec *httptest.ResponseRecorder) {
+				var resp struct {
+					Data []*validatorTrace `json:"data"`
+				}
+				require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &resp))
+				require.Len(t, resp.Data, 1)
+				assert.Equal(t, phase0.Slot(150), resp.Data[0].Slot)
+				assert.Equal(t, "PROPOSER", resp.Data[0].Role)
+				assert.Equal(t, phase0.ValidatorIndex(1), resp.Data[0].Validator)
+			},
+		},
+		{
+			name: "invalid request - no pubkeys or indices",
+			request: map[string]any{
+				"from":  100,
+				"to":    200,
+				"roles": []string{"PROPOSER"},
+			},
+			setupMock:      func(store *mockTraceStore, validatorStore *mockValidatorStore) {},
+			expectedStatus: http.StatusBadRequest,
+			validateResp: func(t *testing.T, rec *httptest.ResponseRecorder) {
+				var resp struct {
+					Status  string `json:"status"`
+					Message string `json:"error"`
+				}
+				require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &resp))
+				assert.Equal(t, "either pubkeys or indices is required", resp.Message)
+			},
+		},
+		{
+			name: "invalid request - validator not found",
+			request: map[string]any{
+				"from":    100,
+				"to":      200,
+				"roles":   []string{"PROPOSER"},
+				"indices": []uint64{1},
+			},
+			setupMock: func(store *mockTraceStore, validatorStore *mockValidatorStore) {
+				validatorStore.ValidatorByIndexFunc = func(index phase0.ValidatorIndex) (*ssvtypes.SSVShare, bool) {
+					return nil, false
+				}
+			},
+			expectedStatus: http.StatusBadRequest,
+			validateResp: func(t *testing.T, rec *httptest.ResponseRecorder) {
+				var resp struct {
+					Status  string `json:"status"`
+					Message string `json:"error"`
+				}
+				require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &resp))
+				assert.Equal(t, "validator not found: 1", resp.Message)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			store := newMockTraceStore()
+			validatorStore := newMockValidatorStore()
+			tt.setupMock(store, validatorStore)
+
+			exporter := &Exporter{
+				TraceStore: store,
+				Validators: validatorStore,
+			}
+
+			reqBody, err := json.Marshal(tt.request)
+			require.NoError(t, err)
+
+			req := httptest.NewRequest(http.MethodPost, "/validator/traces", strings.NewReader(string(reqBody)))
```

Comment:
nit: the endpoint is not really `/validator/traces`, but `/traces/validator`. It's a small thing but I was actually confused and started to look for that endpoint in `server.go`